compile.c

```diff
@@ -3251,6 +3251,7 @@ iseq_specialized_instruction(rb_iseq_t *iseq, INSN *iobj)
 		  case idLength: SP_INSN(length); return COMPILE_OK;
 		  case idSize:	 SP_INSN(size);	  return COMPILE_OK;
 		  case idEmptyP: SP_INSN(empty_p);return COMPILE_OK;
+		  case idNilP:   SP_INSN(nil_p);return COMPILE_OK;
```

Comment:
```suggestion
		  case idNilP:   SP_INSN(nil_p);  return COMPILE_OK;
```

Upstream might complain without this.